### PR TITLE
clear invalid repo data by commond

### PR DESCRIPTION
**Pull request by zMingGit**

### CODE REVIEW

File: seahub/base/management/commands/clear_invalid_repo_data.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+        self.clear_table('RevisionTags', RevisionTags)
+
+        self.clear_table('PersonalWiki', PersonalWiki)
+
+        self.clear_table('GroupWiki', GroupWiki)
+
+        self.clear_table('UserStarredFiles', UserStarredFiles)
+
+        self.stdout.writeline('Start to clear UserNotification Table')
```
Comment (freeplant) [inline seahub/base/management/commands/clear_invalid_repo_data.py]: UserNotification 应该按照时间来清理就可以，不需要在这里清理

File: seahub/base/management/commands/clear_invalid_repo_data.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        user_notification = UserNotification.objects.all()
+        for detail in all_repo_dumps:
+            user_notification.objects.exclude(detail__contains=detail)
+        user_notification.delete()
+        self.stdout.writeline('UserNotification table has been clear')
+
+        self.clear_table('ExtraGroupsSharePermission', ExtraGroupsSharePermission)
+        self.clear_table('ExtraSharePermission', ExtraSharePermission)
+
+        self.stdout.writeline('Start to clear OrgFileShare/FileShare Table')
```
Comment (freeplant) [inline seahub/base/management/commands/clear_invalid_repo_data.py]: OrgFileShare/FileShare is not used anymore, don't need to be cleared here.

File: seahub/base/management/commands/clear_invalid_repo_data.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    def handle(self, *args, **kwargs):
+        self.stdout.writeline('Started to get all exists repo')
+        self.all_repo= [repo.repo_id for repo in seafile_api.get_repo_list(-1, -1)]
+        trash_repo = [repo.repo_id for repo in seafile_api.get_trash_repo_list(-1, -1)]
+        self.all_repo.extend(trash_repo)
+        self.stdout.writeline('Successly get all exists repo')
+
+        #on_delete is  CASCADE, so FileTag/FileComment will be deleted
+        self.clear_table('FileUUIDMap', FileUUIDMap)
```
Comment (xiez) [inline seahub/base/management/commands/clear_invalid_repo_data.py]: 用列表保存需要清理的表的model，在循环里做。方便后续扩展

File: seahub/base/management/commands/clear_invalid_repo_data.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+from seahub.wiki.models import PersonalWiki, GroupWiki
+from seahub.share.models import ExtraGroupsSharePermission, FileShare, \
+        OrgFileShare, ExtraSharePermission, UploadLinkShare
+
+
+class Command(BaseCommand):
+    help = "Clear invalid data when repo deleted"
+
+    def handle(self, *args, **kwargs):
+        self.stdout.writeline('Started to get all exists repo')
```
Comment (xiez) [inline seahub/base/management/commands/clear_invalid_repo_data.py]: start to ..

File: seahub/base/management/commands/clear_invalid_repo_data.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        self.clear_table('GroupWiki', GroupWiki)
+
+        self.clear_table('UserStarredFiles', UserStarredFiles)
+
+        self.clear_table('ExtraGroupsSharePermission', ExtraGroupsSharePermission)
+        self.clear_table('ExtraSharePermission', ExtraSharePermission)
+
+        self.clear_table('UploadLinkShare', UploadLinkShare)
+
+        self.stdout.writeline('All invalid repo data deleted')
```
Comment (xiez) [inline seahub/base/management/commands/clear_invalid_repo_data.py]: are deleted

**Pull request merged**